packages/build/src/plugins_core/secrets_scanning/utils.ts

```diff
@@ -53,36 +89,56 @@ export function isSecretsScanningEnabled(env: Record<string, unknown>): boolean
  * @returns string[]
  */
 export function getSecretKeysToScanFor(env: Record<string, unknown>, secretKeys: string[]): string[] {
-  let omitKeys: string[] = []
-  if (typeof env.SECRETS_SCAN_OMIT_KEYS === 'string') {
-    omitKeys = env.SECRETS_SCAN_OMIT_KEYS.split(',')
-      .map((s) => s.trim())
-      .filter(Boolean)
-  }
+  const filteredSecretKeys = filterOmittedKeys(env, secretKeys)
+  return filteredSecretKeys.filter((key) => isValueNonTrivial(env[key]))
+}
 
-  return secretKeys.filter((key) => {
-    if (omitKeys.includes(key)) {
-      return false
-    }
+/**
+ * given the explicit secret keys and env vars, return the list of non-secret keys which should be scanned in the enhanced secret scan
+ * (i.e. any that look very likely to be secrets).
+ * This will also filter out keys passed in the SECRETS_SCAN_OMIT_KEYS env var.
+ *
+ * @param env env vars list
+ * @param secretKeys
+ * @returns string[]
+ */
+export function getNonSecretKeysToScanFor(env: Record<string, unknown>, secretKeys: string[]): string[] {
+  const nonSecretKeys = Object.keys(env).filter((key) => !secretKeys.includes(key))
+  const filteredNonSecretKeys = filterOmittedKeys(env, nonSecretKeys)
 
+  const nonSecretKeysToScanFor = filteredNonSecretKeys.filter((key) => {
     const val = env[key]
-    if (typeof val === 'string') {
-      // string forms of booleans
-      if (val === 'true' || val === 'false') {
-        return false
-      }
-
-      // non-trivial/non-empty values only
-      return val.trim().length > 4
-    } else if (typeof val === 'boolean') {
-      // booleans are trivial values
+    if (!isValueNonTrivial(val)) {
       return false
-    } else if (typeof val === 'number' || typeof val === 'object') {
-      return JSON.stringify(val).length > 4
     }
-
-    return !!val
+    return isLikelySecretValue(val)
   })
+  return nonSecretKeysToScanFor
+}
+
+const AWS_PREFIXES = ['aws_', 'asia']
+const SLACK_PREFIXES = ['xoxb-', 'xwfp-', 'xoxb-', 'xoxp-', 'xapp-']
+const LIKELY_SECRET_PREFIXES = ['pk_', 'sk_', 'pat_', 'db_', 'github_pat_', ...AWS_PREFIXES, ...SLACK_PREFIXES]
+const LIKELY_SECRET_MIN_LENGTH = 16
+
+/**
+ * When the enhanced secret scan is run, we check any env vars _not_ marked as secret if they are highly likely to be secret values.
+ * For now, this means the value is a string of at least 16 chars and starts with one of the known prefixes.
+ *
+ * @param val env var value
+ * @returns boolean
+ */
+function isLikelySecretValue(val): boolean {
```

Comment:
This is pretty simplistic at the moment, but the idea is to catch the low-hanging fruit for now and build on it more later